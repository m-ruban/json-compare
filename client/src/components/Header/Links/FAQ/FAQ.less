.faq-link:hover {
  cursor: pointer;
}
