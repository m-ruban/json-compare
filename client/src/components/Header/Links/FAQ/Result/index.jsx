import React from 'react';
import { Badge } from 'reactstrap';
import trl from 'modules/translation';

const Result = () => (
    <div>
        <ol>
            <li>
                <Badge color="req" pill>
                    !
                </Badge>
                {trl('FAQ.ComparisonResult.required')}
            </li>
            <li>
                <Badge color="diff" pill>
                    !
                </Badge>
                {trl('FAQ.ComparisonResult.difference')}
            </li>
        </ol>
    </div>
);

export default Result;
