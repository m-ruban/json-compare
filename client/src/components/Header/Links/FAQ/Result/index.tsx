import React from 'react';

import Mark from 'components/Mark';
import trl from 'modules/translation';

const Result = () => (
    <div>
        <ol>
            <li>
                <Mark type="req" text="!" pill />
                {trl('FAQ.ComparisonResult.required')}
            </li>
            <li>
                <Mark type="diff" text="!" pill />
                {trl('FAQ.ComparisonResult.difference')}
            </li>
        </ol>
    </div>
);

export default Result;
