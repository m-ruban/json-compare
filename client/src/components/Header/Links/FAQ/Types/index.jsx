import React from 'react';
import trl from 'modules/translation';
import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';

const Types = () => (
    <div>
        {trl('FAQ.ComparisonTypes.Answer.text')}
        <ol>
            <li>
                {EQUALITY_TYPE_STRICT} - {trl('FAQ.ComparisonTypes.Answer.noStrict')}
            </li>
            <li>
                {EQUALITY_TYPE_NOT_STRICT} - {trl('FAQ.ComparisonTypes.Answer.strict')}
            </li>
            <li>
                {EQUALITY_TYPE_IGNORE} - {trl('FAQ.ComparisonTypes.Answer.ignore')}
            </li>
        </ol>
    </div>
);

export default Types;
