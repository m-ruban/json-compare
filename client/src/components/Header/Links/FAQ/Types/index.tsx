import React from 'react';

import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';
import trl from 'modules/translation';

const Types = () => (
    <div>
        {trl('FAQ.ComparisonTypes.Answer.text')}
        <ol>
            <li>
                {EQUALITY_TYPE_STRICT} - {trl('FAQ.ComparisonTypes.Answer.noStrict')}
            </li>
            <li>
                {EQUALITY_TYPE_NOT_STRICT} - {trl('FAQ.ComparisonTypes.Answer.strict')}
            </li>
            <li>
                {EQUALITY_TYPE_IGNORE} - {trl('FAQ.ComparisonTypes.Answer.ignore')}
            </li>
        </ol>
    </div>
);

export default Types;
