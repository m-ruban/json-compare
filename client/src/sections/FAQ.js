import React from 'react';
import { Row, Col, ListGroup, ListGroupItem, ListGroupItemHeading, ListGroupItemText, Badge } from 'reactstrap';
import withContainer from 'hocs/withContainer';

const Faq = () => (
    <Row>
        <Col xs={8}>
            <ListGroup>
                <ListGroupItem>
                    <ListGroupItemHeading>For what?</ListGroupItemHeading>
                    <ListGroupItemText>
                        This tool is being used for studying the <b>JSON</b> object hierarchy and comparing two{' '}
                        <b>JSON</b> objects.
                        <br />
                        Also, in the presence of the possibility of comparing two <b>JSON</b> objects.
                    </ListGroupItemText>
                </ListGroupItem>
                <ListGroupItem>
                    <ListGroupItemHeading>Comparison types</ListGroupItemHeading>
                    <ListGroupItemText>
                        The application has three types of comparison:
                        <br />
                        1) <b>==</b> - the application uses no strict equality (13 == &apos;13&apos;)
                        <br />
                        2) <b>===</b> - the application uses strict equality (13 !== &apos;13&apos;)
                        <br />
                        3) <b>Ignore</b> - the application ignore value of the key
                    </ListGroupItemText>
                </ListGroupItem>
                <ListGroupItem>
                    <ListGroupItemHeading>Comparison results</ListGroupItemHeading>
                    <ListGroupItemText>
                        <Badge color="req" pill>
                            !
                        </Badge>{' '}
                        - required key
                        <br />
                        <Badge color="diff" pill>
                            !
                        </Badge>{' '}
                        - the key exists, but it&apos;s value doesn&apos;t match the required one
                        <br />
                    </ListGroupItemText>
                </ListGroupItem>
            </ListGroup>
        </Col>
    </Row>
);

export default withContainer(Faq);
