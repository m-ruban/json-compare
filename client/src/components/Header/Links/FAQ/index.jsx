import React, { useState, useCallback, Fragment } from 'react';
import {
    Button,
    Modal,
    ModalHeader,
    ModalBody,
    ModalFooter,
    ListGroupItem,
    ListGroupItemHeading,
    ListGroupItemText,
    Badge,
    ListGroup,
} from 'reactstrap';

import trl from 'modules/translation';
import { EQUALITY_TYPE_IGNORE, EQUALITY_TYPE_NOT_STRICT, EQUALITY_TYPE_STRICT } from 'constants/equality';
import 'components/Header/Links/FAQ/FAQ.less';

const FAQ = () => {
    const [modal, setModal] = useState(false);
    const toggle = useCallback(() => {
        setModal(!modal);
    }, [modal]);

    return (
        <Fragment>
            <span className="faq-link" onClick={toggle}>
                {trl('Header.Links.FAQ.title')}
            </span>
            <Modal isOpen={modal} toggle={toggle} size="lg">
                <ModalHeader toggle={toggle}>FAQ</ModalHeader>
                <ModalBody>
                    <ListGroup>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ForWhat.question')}</ListGroupItemHeading>
                            <ListGroupItemText>{trl('FAQ.ForWhat.answer')}</ListGroupItemText>
                        </ListGroupItem>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ComparisonTypes.question')}</ListGroupItemHeading>
                            <div>
                                {trl('FAQ.ComparisonTypes.Answer.text')}
                                <ol>
                                    <li>
                                        {EQUALITY_TYPE_STRICT} - {trl('FAQ.ComparisonTypes.Answer.noStrict')}
                                    </li>
                                    <li>
                                        {EQUALITY_TYPE_NOT_STRICT} - {trl('FAQ.ComparisonTypes.Answer.strict')}
                                    </li>
                                    <li>
                                        {EQUALITY_TYPE_IGNORE} - {trl('FAQ.ComparisonTypes.Answer.ignore')}
                                    </li>
                                </ol>
                            </div>
                        </ListGroupItem>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ComparisonResult.question')}</ListGroupItemHeading>
                            <div>
                                <ol>
                                    <li>
                                        <Badge color="req" pill>
                                            !
                                        </Badge>
                                        {trl('FAQ.ComparisonResult.required')}
                                    </li>
                                    <li>
                                        <Badge color="diff" pill>
                                            !
                                        </Badge>
                                        {trl('FAQ.ComparisonResult.difference')}
                                    </li>
                                </ol>
                            </div>
                        </ListGroupItem>
                    </ListGroup>
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" onClick={toggle}>
                        {trl('Header.Links.FAQ.button')}
                    </Button>
                </ModalFooter>
            </Modal>
        </Fragment>
    );
};

export default FAQ;
