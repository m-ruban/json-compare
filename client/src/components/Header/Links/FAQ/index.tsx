import React from 'react';
import {
    Button,
    ListGroup,
    ListGroupItem,
    ListGroupItemHeading,
    Modal,
    ModalBody,
    ModalFooter,
    ModalHeader,
} from 'reactstrap';

import 'components/Header/Links/FAQ/FAQ.less';
import Result from 'components/Header/Links/FAQ/Result';
import Types from 'components/Header/Links/FAQ/Types';
import trl from 'modules/translation';

const FAQ = () => {
    const [modal, setModal] = React.useState(false);
    const toggle = React.useCallback(() => {
        setModal(!modal);
    }, [modal]);

    return (
        <>
            <span className="faq-link" onClick={toggle}>
                {trl('Header.Links.FAQ.title')}
            </span>
            <Modal isOpen={modal} toggle={toggle} size="lg">
                <ModalHeader toggle={toggle}>FAQ</ModalHeader>
                <ModalBody>
                    <ListGroup>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ForWhat.question')}</ListGroupItemHeading>
                            <div>{trl('FAQ.ForWhat.answer')}</div>
                        </ListGroupItem>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ComparisonTypes.question')}</ListGroupItemHeading>
                            <Types />
                        </ListGroupItem>
                        <ListGroupItem>
                            <ListGroupItemHeading>{trl('FAQ.ComparisonResult.question')}</ListGroupItemHeading>
                            <Result />
                        </ListGroupItem>
                    </ListGroup>
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" onClick={toggle}>
                        {trl('Header.Links.FAQ.button')}
                    </Button>
                </ModalFooter>
            </Modal>
        </>
    );
};

export default FAQ;
