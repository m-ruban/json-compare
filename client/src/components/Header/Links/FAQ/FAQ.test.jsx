import React from 'react';
import { render } from 'enzyme';
import FAQ from 'components/Header/Links/FAQ';

describe('<FAQ />', () => {
    it('Renders correctly', () => {
        const output = render(<FAQ />);
        expect(output).toMatchSnapshot();
    });
});
