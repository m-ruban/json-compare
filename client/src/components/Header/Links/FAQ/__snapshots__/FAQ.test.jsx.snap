// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQ /> Renders correctly 1`] = `
<span
  class="faq-link"
>
  FAQ
</span>
`;
